### PR TITLE
Update artifact actions to v4 #168

### PR DESCRIPTION
**Pull request by javierfreire**

### CODE REVIEW

File: src/action.ts [hunk 1]
```diff
@@ -326,14 +326,9 @@ class Action {
...
-      const artifactClient = artifact.create()
+      const artifactClient = new artifact.DefaultArtifactClient()
       const artifactName = await this.getArtifactName(executionGraphId)
-      
-      const uploadResult = await artifactClient.uploadArtifact(artifactName, artifacts, baseDir, { continueOnError: true })
-      
-      if (uploadResult.failedItems.length > 0) {
-        core.warning(`The following files could not be uploaded: ${uploadResult.failedItems}`)
-      }
+      await artifactClient.uploadArtifact(artifactName, artifacts, baseDir)
```
Comment (mpermar) [inline src/action.ts:331]: Can we keep the other functionality somehow? I mean continuing on errors and warning on failed uploads.

Comment (javierfreire) [reply src/action.ts:335]: With the v4 all the artifacts are uploaded packed as one file, so that check makes no sense.

Comment (mpermar) [reply src/action.ts:335]: Ok. I see. We need to check for upload errors then. Right now, I believe a glitch on GitHub uploads does not break the action. But with your change, [looking at the client library code](https://github.com/actions/toolkit/blob/77f247b2f3e5d82ecd0e27573ef30c75d5d9a2cb/packages/artifact/src/internal/client.ts#L134), a glitch will show a warning but also bubbles up the error which I presume will make our action to fail. 

Can you see if this make sense? If it does, think can we can avoid breaking. Overall, artifact upload is a nice to have, if it fails we still want to have the action succeed.

Comment (javierfreire) [reply src/action.ts]: Done, I added a try/catch to handle the error

**Comment by javierfreire** [comment]
This PR is replaced by the [PR](https://github.com/vmware-labs/vmware-image-builder-action/pull/171)

**Pull request closed without merging**